test/handshake_helper.c

```diff
@@ -266,6 +300,16 @@ HANDSHAKE_RESULT do_handshake(SSL_CTX *server_ctx, SSL_CTX *client_ctx,
     ret.client_alert_received = server_ex_data.alert_received;
     ret.server_protocol = SSL_version(server);
     ret.client_protocol = SSL_version(client);
+    ret.servername = ((SSL_get_SSL_CTX(server) == server_ctx)
+                      ? SSL_TEST_SERVERNAME_SERVER1
+                      : SSL_TEST_SERVERNAME_SERVER2);
+    if ((sess = SSL_get0_session(client)) != NULL)
+        SSL_SESSION_get0_ticket(sess, &tick, &len);
+    if (tick == NULL || len == 0)
+        ret.session_ticket_expected = SSL_TEST_SESSION_TICKET_EXPECTED_NO;
```

Comment:
nit: perhaps call the enums SSL_TEST_SESSION_TICKET_NO etc, or SSL_TEST_SESSION_TICKET_RECEIVED_NO (i.e., omit EXPECTED)? 

Then, you can also call the field 'ret.session_ticket' or 'ret.session_ticket_received' accordingly.

(It's just a little confusing to set an expectation on a result.)